src/lib/components/DeleteButton.js

```diff
@@ -32,10 +27,15 @@ export default class DeleteButtonComponent extends Component {
     const {
       isSaved,
       isPublished,
-      handleClick,
+      deleteClick,
       ...uiProps // only has ActionButton props
     } = this.props;
 
+    const handleDelete = (event, formik) => {
+      deleteClick(event, formik);
+      this.hanldeClose();
+    };
+
     const action = isPublished ? 'discard' : 'delete';
     const titleizedAction = action[0].toUpperCase() + action.slice(1);
```

Comment:
minor: it would be nice to add a small comment e.g "Capitalize first letter" or create a function `capitalizeFirstLetter(action)` just to be easier to read that line :)

Reply:
Good idea. I will wrap it in a module function for now and perhaps we will see over time its need elsewhere and we can gradually make it accessible to more code.